src/libraries/TickBitmap.sol

```diff
@@ -28,11 +28,25 @@ library TickBitmap {
     /// @param tick The tick to flip
     /// @param tickSpacing The spacing between usable ticks
     function flipTick(mapping(int16 => uint256) storage self, int24 tick, int24 tickSpacing) internal {
-        unchecked {
-            if (tick % tickSpacing != 0) revert TickMisaligned(tick, tickSpacing); // ensure that the tick is spaced
-            (int16 wordPos, uint8 bitPos) = position(tick / tickSpacing);
-            uint256 mask = 1 << bitPos;
-            self[wordPos] ^= mask;
+        /// @solidity memory-safe-assembly
+        assembly {
+            // ensure that the tick is spaced
+            if smod(tick, tickSpacing) {
```

Comment:
i actually think a comment of just the whole old function might be good - so that its apparent what wordPos and bitPos are in this function given it no longer calls `position`